### ISSUE TITLE
Potential checkerboard artifacts in terramind_v1_tokenizer_s1grd reconstruction

### ISSUE THREAD

**Issue by Luo-Z13** [IBM/terramind#35]
Hi TerraMind Team,

First of all, thank you for the impressive work on the TerraMind foundation models!

I am currently evaluating the S1-GRD Tokenizer and noticed significant checkerboard/grid artifacts in the reconstructed output when using decode_quant.

**Observation:**
While the s1rtc tokenizer works perfectly on the provided example images, the s1grd tokenizer consistently exhibits a regular grid pattern across the entire reconstructed image. This pattern is particularly visible in the continuous latent reconstruction (decode_quant).

**Steps taken to rule out data issues:**

- Clipping: I clipped the input SAR values to match the pretraining statistics (approx. -35 to 2 dB).
- Normalization: I used the official v1_pretraining_mean/std for untok_sen1grd@224.
- Resize test: To see if the issue was related to input texture, I resized a 128x128 image to 224x224. As shown in the attached image, the checkerboard artifacts persist.

Since there is currently no official S1-GRD example image in the repository, could you please:

1. Confirm if this is a known behavior or if I am missing a specific preprocessing step (e.g., a specific speckle filter or log-scaling)?
2. If possible, provide a sample S1-GRD image that demonstrates a clean reconstruction, so I can verify my local implementation?

```
import torch
import numpy as np
import rasterio
import matplotlib.pyplot as plt
from pathlib import Path
import torch.nn.functional as F
from terratorch import FULL_MODEL_REGISTRY
from terratorch.models.backbones.terramind.model.terramind_register import v1_pretraining_mean, v1_pretraining_std

def run_reconstruction(input_path):
    input_path = Path(input_path)
    device = "cuda" if torch.cuda.is_available() else "cpu"
    
    model_type = "terramind_v1_tokenizer_s1grd"
    modality_key = 'untok_sen1grd@224'
    model = FULL_MODEL_REGISTRY.build(model_type, pretrained=True).to(device).eval()

    if input_path.suffix == '.npy':
        input_np = np.load(input_path)
        if input_np.ndim == 4: input_np = input_np[0]
        sar_data = input_np[:2, :, :]
    else:
        with rasterio.open(input_path) as src:
            sar_data = src.read([1, 2])

    data = torch.from_numpy(sar_data).float().to(device).unsqueeze(0)
    data = torch.nan_to_num(data, nan=-30.0, posinf=0.0, neginf=-30.0)
    data = torch.clamp(data, min=-35.0, max=2.0)

    mean = torch.tensor(v1_pretraining_mean[modality_key], device=device)[None, :, None, None]
    std = torch.tensor(v1_pretraining_std[modality_key], device=device)[None, :, None, None]
    input_normalized = (data - mean) / std

    with torch.no_grad():
        quant, _, tokens = model.encode(input_normalized)
        recon_tokens = model.decode_tokens(tokens, timesteps=10)
        recon_quant = model.decode_quant(quant, timesteps=10)

    recon_tokens_dn = (recon_tokens * std) + mean
    recon_quant_dn = (recon_quant * std) + mean

    def process_sar_for_vis(tensor_data):
        img = tensor_data.squeeze(0).cpu().numpy()
        vv, vh = img[0], img[1]
        ratio = vv / (vh + 1e-6)
        def scale(ch):
            vmin, vmax = np.percentile(ch, (2, 98))
            return np.clip((ch - vmin) / (vmax - vmin + 1e-6), 0, 1)
        return np.stack([scale(vv), scale(vh), scale(ratio)], axis=-1)

    fig, axes = plt.subplots(1, 3, figsize=(24, 8))
    axes[0].imshow(process_sar_for_vis(data))
    axes[0].set_title(f"Original SAR\n{input_path.name}")
    axes[0].axis('off')
    axes[1].imshow(process_sar_for_vis(recon_tokens_dn))
    axes[1].set_title("Recon from Tokens (Discrete)")
    axes[1].axis('off')
    axes[2].imshow(process_sar_for_vis(recon_quant_dn))
    axes[2].set_title("Recon from Quant (Continuous)")
    axes[2].axis('off')

    plt.tight_layout()
    plt.savefig(f"issue_repro_{input_path.stem}.png", dpi=150)
    plt.show()

if __name__ == "__main__":
    run_reconstruction("S1A_10000.npy")

```

<img width="3808" height="1298" alt="Image" src="https://github.com/user-attachments/assets/95aeb3d3-cdd3-4432-81c3-178a48ab115d" />

**Comment by blumenstiel** [comment]
Hi @Luo-Z13, it might be the wrong image size when using `decode_quant`. In `decode_tokens` we have the following code:
```
        image_size = (kwargs.pop("image_size", None) or
                      (tokens.shape[-2] * self.patch_size, tokens.shape[-1] * self.patch_size))

        dec = self.decode_quant(quant, image_size=image_size, **kwargs)
```

If you only run `decode_quant` without providing `image_size`, the model assumes the default which is 256. I assume your image has a different shape?

**Comment by blumenstiel** [comment]
RTC is coming from Planetary Computer, see TerraMesh paper for details.
